### ISSUE TITLE
Reaching ets limit when trying out debugger

### ISSUE THREAD

**Issue by nulian** [KronicDeth/intellij-elixir#677]
When I run phoenix.server as a debugger I get these set of errors

```
** Invalid beam file or no abstract code: 'Elixir.Phoenix.HTML.Safe'

10:25:40.551 [error] ** Too many db tables **


** Invalid beam file or no abstract code: 'Elixir.Phoenix.HTML.Tag'

10:25:40.553 [error] ** Too many db tables **


** Invalid beam file or no abstract code: 'Elixir.Phoenix.HTML'

10:25:40.554 [error] ** Too many db tables **


** Invalid beam file or no abstract code: 'Elixir.Phoenix.LiveReload.Channel'

10:25:40.557 [error] ** Too many db tables **


** Invalid beam file or no abstract code: 'Elixir.Phoenix.LiveReloader.Socket'

10:25:40.558 [error] ** Too many db tables **


** Invalid beam file or no abstract code: 'Elixir.Phoenix.LiveReloader'

10:25:40.561 [error] ** Too many db tables **


** Invalid beam file or no abstract code: 'Elixir.Phoenix.PubSub.BroadcastError'

10:25:40.562 [error] ** Too many db tables **


** Invalid beam file or no abstract code: 'Elixir.Phoenix.PubSub.GC'

10:25:40.566 [error] ** Too many db tables **
```
Which I think is because of reaching ets table limits. I tried it out on a fairly large project(391 files just in lib folder) that makes fair amount of use of ets tables.

**Comment by nulian** [comment]
With the following deps list

```
defp deps do
    [
      {:phoenix, "~> 1.2.1"},
      {:phoenix_ecto, "~> 3.2.3"},
      {:ecto, "~> 2.1.4", override: true},
      {:phoenix_pubsub, "~> 1.0"},
      {:mariaex, "~> 0.8.1"},
      {:sbroker, "~> 1.0.0"},
      {:phoenix_html, "~> 2.3"},
      {:phoenix_live_reload, "~> 1.0", only: :dev},
      {:phoenix_pubsub, "~> 1.0"},
      {:cowboy, "~> 1.0"},
      {:gproc, "~> 0.6.1"},
      {:exrm, ">= 0.16.0", warn_missing: false},
      {:edeliver, github: "benrom/edeliver"},
      {:roger, github: "bettyblocks/roger", tag: "1.0.0-beta9"},
      {:singleton, "~> 1.0"},
      {:credo, "~> 0.5", only: [:dev, :test]},
      {:excoveralls, "~> 0.4", only: :test},
      {:cors_plug, ">= 1.1.0"},
      {:ex_doc, "~> 0.11", only: :dev},
      {:inflex, "~> 1.7", override: true},
      {:earmark, ">= 0.0.0", only: :dev},
      {:number, "~> 0.4"},
      {:exiban, "~> 0.0.5"},
      {:appsignal, "~> 1.0"},
      {:json_web_token, "~> 0.2"},
      {:timex, github: "bitwalker/timex", override: true},
      {:timex_ecto, "~> 3.1.1"},
      {:liquid, "~> 0.6.0"},
      {:html_sanitize_ex, "~> 1.0"},
      {:sql_dust, "~> 0.3.10"},
      {:redix, "~> 0.5"},
      {:redix_pubsub, "~> 0.2"},
      {:logger_papertrail_backend, "~> 0.1.0"},
      {:relx, github: "erlware/relx", override: true},
      {:sweet_xml, "~> 0.6.1"},
      {:exml, "~>0.1.0"},
      {:exprintf, "~> 0.2.1"},
      {:poison, "~> 3.1", override: true},
      {:mogrify, "~> 0.5.1"},
      {:ex_aws, github: "squareduck/ex_aws"},
      {:image_optimizer_ex, github: "squareduck/image-optimizer-ex"},
      {:compiler_cache, "~> 1.1"},
      {:jsx, "~> 2.8.1"},
      {:httpoison, "~> 0.9"},
      {:instream, "~> 0.14"},
      {:deep_merge, "~> 0.1.0"},
      {:temp, "~> 0.4"},
      {:bypass, "~> 0.1", only: :test}
    ]
  end
```

**Comment by KronicDeth** [comment]
The Elixir code in IntelliJ Elixir itself, these two files:
* https://github.com/KronicDeth/intellij-elixir/blob/master/resources/debugger/lib/debug_server.ex
* https://github.com/KronicDeth/intellij-elixir/blob/master/resources/debugger/lib/debug_task.ex

Do not explicitly create ETS tables.  If ETS tables are being created, it's indirectly due to how the Erlang libs that the Elixir code is using:
* `:int.interable`
* `:int.ni`
* `:int.auto_attach`
* `:int.interpreted`
* `:int.delete_break`
* `:int.step`
* `:int.next`
* `:int.finish`
* `:int.continue`
* `:dbg_iserver.safe_call`
* `:int.meta`

Most of those calls you can use manually as outlined in this article: http://blog.plataformatec.com.br/2016/04/debugging-techniques-in-elixir-lang/.  If you still have the ETS problem when using the calls manually, then it's an issue with Erlang libraries and/or their interaction with your code.

1. Could it be that when certain processes in your project are paused, other processes start to build up an excess of ETS tables on their own because the paused processes aren't draining the ETS producers? 
2. Have you hit the ETS problem when using `require IEx; IEx.pry` style debugging?

**Comment by nulian** [comment]
Normally I don't have any problems with using IEx.pry though not all parts are as easy to debug because of the supervisors.
But even if I disable the entire supervisor tree in the start function it gives those errors. It looks like when it's trying to :int.ni all the modules in the project.

And that is probably also causing another problem when I increase the ETS tables limit and succeed starting it. Because everything is bound in the debugger including the libraries it slows down the project so much I get problems with our supervisors timing out. Though with increasing the timeouts I can get the project to work with the debugger though startup takes about 50 seconds on the first request. 

When I manually use the :int.ni I don't have that problem but it's probably because everything is being debugged now.

**Comment by KronicDeth** [comment]
I think the way around this is to start porting in some of the UI from `:debugger`, such as the ability to disable `:int.ni` of certain module and to show which modules are currently being interpreted, so you can manually tune the list.

**Comment by KronicDeth** [comment]
* Allow modules to not be interpreted was in 7c1f8dec1e043eaf1149931e4e12febd4d57cb9a which was first released in [v7.4.0](https://github.com/KronicDeth/intellij-elixir/releases/tag/v7.4.0) in March 2018
* Showing modules being interpreted was in 9fd9ee9c7e001d79da9b789f9e34e0d9a765f8eb which was first released in [v8.0.0](https://github.com/KronicDeth/intellij-elixir/releases/tag/v8.0.0) in August 2018
